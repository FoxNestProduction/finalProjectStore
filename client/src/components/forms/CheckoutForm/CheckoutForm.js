/* eslint-disable max-len */
import React, { memo, useEffect, useState } from 'react';
import { useNavigate } from 'react-router';
import { shallowEqual, useDispatch, useSelector } from 'react-redux';
import { Field, Form, Formik } from 'formik';

import Box from '@mui/material/Box';
import Stack from '@mui/material/Stack';
import Divider from '@mui/material/Divider';
import FormControlLabel from '@mui/material/FormControlLabel';
import MenuItem from '@mui/material/MenuItem';
import Radio from '@mui/material/Radio';
import RadioGroup from '@mui/material/RadioGroup';
import Typography from '@mui/material/Typography';
import FormControl from '@mui/material/FormControl';
import InputLabel from '@mui/material/InputLabel';
import InputMask from 'react-input-mask';

import Input from '../../inputs/Input/Input';
import validationSchema from './validationSchema';
import SelectForFormik from '../../inputs/Select/Select';
import CheckoutActions from './CheckoutActions';
import {
  subtitle,
  paymentRadioBtn, paymentWrapper,
} from './styles';
import { updateCustomer } from '../../../redux/slices/userSlice';
import { CHECKOUT_SS_KEY } from '../../../constants/constants';
import {
  getDataFromSessionStorage,
  removeDataFromSessionStorage,
  updateSessionStorageValues,
} from '../../../utils/sessionStorageHelpers';
import { putNewOrder, setPendingOrderInfo } from '../../../redux/slices/orderSlice';
import saveUserInfoToSessionStorage from '../../../utils/saveUserInfoToSessionStorage';
import { deleteCart, resetCart } from '../../../redux/slices/cartSlice';

const CheckoutForm = () => {
  const navigate = useNavigate();
  const dispatch = useDispatch();

  const getInitialValues = () => ({
    name: '',
    email: '',
    tel: '',
    city: 'Kyiv',
    street: '',
    house: '',
    apartment: '',
    payment: 'Card',
  });

  const [initialValues, setInitialValues] = useState(getInitialValues);

  const isUserAuthorized = useSelector((state) => state.authorization.isUserAuthorized);
  const user = useSelector((state) => state.user.user, shallowEqual);
  const token = useSelector((state) => state.authorization.token);
  const cart = useSelector((state) => state.cart.cart.products, shallowEqual);

  const updateCustomerLoading = useSelector((state) => state.user.loading.updatedCustomer);
  const orderLoading = useSelector((state) => state.order.loading);
  const orderError = useSelector((state) => state.order.error);

  useEffect(() => {
    const checkoutValues = getDataFromSessionStorage(CHECKOUT_SS_KEY);
    const newValues = getInitialValues();

    if (checkoutValues) {
      Object.keys(checkoutValues).forEach((key) => {
        if (key in newValues) {
          newValues[key] = checkoutValues[key];
        }
      });
    }
    setInitialValues(newValues);
  }, [isUserAuthorized]);

  const handleFieldBlur = (e, handleBlur) => {
    handleBlur(e);
    updateSessionStorageValues(CHECKOUT_SS_KEY, { [e.target.name]: e.target.value });
  };

  const handleContinue = async (values) => {
    // updating user info in DB and user slice
    if (isUserAuthorized && token) {
      const customerUpdates = {
        telephone: values.tel,
      };
      await dispatch(updateCustomer(customerUpdates));
    }

    const { name, email, tel, city, street, house, apartment, payment } = values;
    const newOrder = {
      products: cart,
      name,
      email,
      mobile: tel,
      deliveryAddress: {
        city,
        street,
        house,
        apartment,
      },
      paymentInfo: payment,
    };

    if (isUserAuthorized && user) {
      const { _id: id } = user;
      newOrder.customerId = id;
    }

    if (values.payment === 'Cash') {
      newOrder.status = 'new_order/cash';
      const response = await dispatch(putNewOrder(newOrder)).unwrap();
      if (response.status === 200) {
        removeDataFromSessionStorage(CHECKOUT_SS_KEY);
        dispatch(resetCart());
        if (isUserAuthorized && user) {
          saveUserInfoToSessionStorage(user);
          dispatch(deleteCart());
        }
        navigate('/order-confirmation');
      }
    } else {
      dispatch(setPendingOrderInfo(newOrder));
      navigate('/checkout/payment');
    }
  };

  const setInitialTouched = () => {
    const values = getDataFromSessionStorage(CHECKOUT_SS_KEY);
    if (values) {
      return {
        name: 'name' in values,
        email: 'email' in values,
        tel: 'tel' in values,
        street: 'street' in values,
        house: 'house' in values,
      };
    }
    return null;
  };

  const setIsValid = (touched, errors) => {
    return !Object.keys(errors).some((key) => touched[key] === true);
  };

  return (
    <Formik
      initialValues={initialValues}
      onSubmit={handleContinue}
      validationSchema={validationSchema}
      enableReinitialize
      initialTouched={setInitialTouched()}
    >
      {({ handleBlur, touched, errors }) => (
        <Form>
          <Stack spacing={4}>
            <Divider />
            <Typography variant="h3" component="h2" align="left" sx={subtitle}>
              Personal Information
            </Typography>

            <Input
              name="name"
              id="checkout-name"
              label="Name*"
              bgColor="#FFF"
              onBlur={(e) => {
                handleFieldBlur(e, handleBlur);
              }}
            />
            <Input
              name="email"
              id="checkout-email"
              label="Email Address*"
              bgColor="#FFF"
              onBlur={(e) => { handleFieldBlur(e, handleBlur); }}
            />

            <Field name="tel">
              {({ field }) => (
                <InputMask
                  mask="+38 (099) 999-99-99"
                  {...field}
                  onBlur={(e) => { handleFieldBlur(e, handleBlur); }}
                >
                  <Input type="tel" name="tel" id="checkout-tel" bgColor="#FFF" label="Phone Number*" />
                </InputMask>
              )}
            </Field>

            <Divider />
            <Typography variant="h3" component="h2" align="left" sx={subtitle}>
              Delivery Information
            </Typography>

            <FormControl fullWidth>
              <InputLabel id="checkout-city-label">City*</InputLabel>
              <Field
                name="city"
                label="City*"
                component={SelectForFormik}
                labelId="checkout-city-label"
                id="checkout-city"
                bgColor="#FFF"
                onBlur={(e) => { handleFieldBlur(e, handleBlur); }}
              >
                <MenuItem value="Kyiv">Kyiv</MenuItem>
                <MenuItem value="Lviv">Lviv</MenuItem>
              </Field>
            </FormControl>

            <Input
              name="street"
              id="checkout-street"
              label="Street*"
              bgColor="#FFF"
              onBlur={(e) => { handleFieldBlur(e, handleBlur); }}
            />
            <Box sx={{ display: 'flex', gap: '5%' }}>
              <Input
                name="house"
                id="checkout-house"
                label="House*"
                bgColor="#FFF"
                onBlur={(e) => { handleFieldBlur(e, handleBlur); }}
              />
              <Input
                name="apartment"
                id="checkout-apartment"
                label="Apartment"
                bgColor="#FFF"
                onBlur={(e) => { handleFieldBlur(e, handleBlur); }}
              />
            </Box>

            <Divider />
            <Typography variant="h3" component="h2" align="left" sx={subtitle}>
              Payment method
            </Typography>

            <Field name="payment">
              {({ field }) => (
                <FormControl sx={paymentWrapper}>
                  <RadioGroup {...field} onBlur={(e) => { handleFieldBlur(e, handleBlur); }}>
                    <FormControlLabel value="Card" control={<Radio />} label="Card" sx={paymentRadioBtn} />
                    <FormControlLabel value="Cash" control={<Radio />} label="Cash" sx={paymentRadioBtn} />
                  </RadioGroup>
                </FormControl>
              )}
            </Field>

          </Stack>
          {orderError && (
            <Box>
              <Typography variant="body1" component="p" sx={{ color: 'text.error', mt: '15px', mb: '-15px' }}>
                {orderError}
              </Typography>
            </Box>
          )}
          <CheckoutActions
            isValid={setIsValid(touched, errors)}
            loading={(orderLoading || updateCustomerLoading) && !orderError}
          />
        </Form>
      )}
    </Formik>
  );
};

export default memo(CheckoutForm);
