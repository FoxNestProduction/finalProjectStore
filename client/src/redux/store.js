import { configureStore, getDefaultMiddleware, combineReducers } from '@reduxjs/toolkit';
// todo: правила для redux-persist
// eslint-disable-next-line import/no-extraneous-dependencies
import storage from 'redux-persist/lib/storage';
// eslint-disable-next-line import/no-extraneous-dependencies
import { persistReducer } from 'redux-persist';
import modalSlice from './slices/modalSlice';
import productsSlice from './slices/productsSlice';
import authorizationSlice from './slices/authorizationSlice';
import userSlice from './slices/userSlice';
import partnersSlice from './slices/partnersSlice';
import searchSlice from './slices/searchSlice';
import errorSlice from './slices/errorSlice';
import reviewsSlice from './slices/reviewsSlice';
import favouriteSlice from './slices/favouriteSlice';

const authPersistConfig = {
  key: 'authorization',
  version: 1,
  storage,
};

const userPersistConfig = {
  key: 'user',
  version: 1,
  storage,
};

const searchPersistConfig = {
  key: 'search',
  version: 1,
  storage,
};
const favouritePersistConfig = {
  key: 'favorites',
  version: 1,
  storage,
};

/*
    todo: - ЗВЕРНІТЬ УВАГУ!
      якщо вам потрібно щоб ваш стейт не зникав після перезавантаження
      то юзайте підхід як authorization і user,
      якщо ж ні то варіант написання як для modal всередині reducer
      Якщо що пишіть @Ihor_Kacher
*/
const reducer = combineReducers({
  error: errorSlice,
  modal: modalSlice,
  partners: partnersSlice,
  authorization: persistReducer(authPersistConfig, authorizationSlice),
  user: persistReducer(userPersistConfig, userSlice),
  products: productsSlice,
  search: persistReducer(searchPersistConfig, searchSlice),
  reviews: reviewsSlice,
  favourites: persistReducer(favouritePersistConfig, favouriteSlice),
});

const middleware = getDefaultMiddleware({
  serializableCheck: false,
});

const store = configureStore({
  reducer,
  middleware,
});

export default store;
