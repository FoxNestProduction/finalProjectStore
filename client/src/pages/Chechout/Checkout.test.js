import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../../redux/store';
import CheckoutPage from './Checkout';
import CheckoutForm from '../../components/forms/CheckoutForm/CheckoutForm';

describe('CheckoutPage', () => {
  test('should render CheckoutPage', () => {

    const { asFragment } = render(
      <Provider store={store}>
        <MemoryRouter>
          <CheckoutPage titleText="Checkout" formComponent={CheckoutForm} />
        </MemoryRouter>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();

    const title = screen.getByText('Checkout');

    expect(title).toBeInTheDocument();
  });
});