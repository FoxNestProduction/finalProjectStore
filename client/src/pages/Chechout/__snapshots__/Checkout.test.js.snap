// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutPage should render CheckoutPage 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-14tzykr-MuiContainer-root"
  >
    <div
      class="MuiBox-root css-79elbk"
    >
      <div
        class="MuiBox-root css-zfnh4d"
      >
        <h1
          class="MuiTypography-root MuiTypography-h2 MuiTypography-alignCenter css-19o8byk-MuiTypography-root"
        >
          Checkout
        </h1>
        <div
          class="MuiBox-root css-wl2was"
        >
          <svg
            fill="none"
            height="360px"
            viewBox="0 0 188 252"
            width="240px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M50.3065 74.1802C50.4985 68.4272 47.5444 62.69 41.067 59.2275C38.5588 57.8867 35.5874 57.3986 32.6611 56.9928C44.5493 56.6696 48.6765 46.8062 50.3307 39.1707C49.7562 46.0188 54.2728 53.4158 63.5029 54.0983C54.3146 58.3833 51.161 66.2464 50.3065 74.1802Z"
              fill-rule="evenodd"
              stroke="#A597FF"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-miterlimit="1.5"
              stroke-width="6.81286"
            />
            <path
              clip-rule="evenodd"
              d="M10.8227 151.317C10.9187 148.44 9.44166 145.572 6.20296 143.841C4.94883 143.17 3.46315 142.926 2 142.723C7.94411 142.562 10.0077 137.63 10.8348 133.812C10.5476 137.236 12.8059 140.935 17.4209 141.276C12.8268 143.419 11.25 147.35 10.8227 151.317Z"
              fill-rule="evenodd"
              stroke="#A597FF"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-miterlimit="1.5"
              stroke-width="3.40643"
            />
            <path
              clip-rule="evenodd"
              d="M152.836 249.418C152.932 246.542 151.455 243.673 148.217 241.942C146.962 241.271 145.477 241.027 144.014 240.824C149.958 240.663 152.021 235.731 152.848 231.913C152.561 235.337 154.82 239.036 159.435 239.377C154.84 241.52 153.264 245.451 152.836 249.418Z"
              fill-rule="evenodd"
              stroke="#A597FF"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-miterlimit="1.5"
              stroke-width="3.40643"
            />
            <path
              clip-rule="evenodd"
              d="M121.029 165.479C121.221 159.726 118.267 153.989 111.79 150.526C109.281 149.186 106.31 148.697 103.384 148.292C115.272 147.968 119.399 138.105 121.053 130.469C120.479 137.318 124.995 144.715 134.226 145.397C125.037 149.682 121.884 157.545 121.029 165.479Z"
              fill-rule="evenodd"
              stroke="#A597FF"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-miterlimit="1.5"
              stroke-width="6.81286"
            />
            <path
              clip-rule="evenodd"
              d="M162.203 58.5458C162.491 49.9163 158.059 41.3106 148.343 36.1169C144.581 34.1057 140.124 33.3734 135.734 32.7649C153.567 32.28 159.758 17.485 162.239 6.03174C161.377 16.3039 168.152 27.3993 181.997 28.4231C168.215 34.8506 163.484 46.6451 162.203 58.5458Z"
              fill-rule="evenodd"
              stroke="#A597FF"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-miterlimit="1.5"
              stroke-width="10.2193"
            />
          </svg>
        </div>
        <form
          action="#"
        >
          <div
            class="MuiStack-root css-8mcjw0-MuiStack-root"
          >
            <hr
              class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
            />
            <h2
              class="MuiTypography-root MuiTypography-h3 MuiTypography-alignLeft css-1nzy6yt-MuiTypography-root"
            >
              Personal Information
            </h2>
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="checkout-name"
              >
                Name*
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1dnlmzz-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-describedby="helper-text"
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="checkout-name"
                  name="name"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-yjsfm1"
                  >
                    <span>
                      Name*
                    </span>
                  </legend>
                </fieldset>
              </div>
              <p
                class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
                id="helper-text"
              />
            </div>
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="checkout-email"
              >
                Email Address*
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1dnlmzz-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-describedby="helper-text"
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="checkout-email"
                  name="email"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-yjsfm1"
                  >
                    <span>
                      Email Address*
                    </span>
                  </legend>
                </fieldset>
              </div>
              <p
                class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
                id="helper-text"
              />
            </div>
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="checkout-tel"
              >
                Phone Number*
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1dnlmzz-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-describedby="helper-text"
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="checkout-tel"
                  name="tel"
                  type="tel"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-yjsfm1"
                  >
                    <span>
                      Phone Number*
                    </span>
                  </legend>
                </fieldset>
              </div>
              <p
                class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
                id="helper-text"
              />
            </div>
            <hr
              class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
            />
            <h2
              class="MuiTypography-root MuiTypography-h3 MuiTypography-alignLeft css-1nzy6yt-MuiTypography-root"
            >
              Delivery Information
            </h2>
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="true"
                id="checkout-city-label"
              >
                City*
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-18r46to-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
              >
                <div
                  aria-controls=":r0:"
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  aria-labelledby="checkout-city-label checkout-city"
                  class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                  id="checkout-city"
                  role="combobox"
                  tabindex="0"
                >
                  Kyiv
                </div>
                <input
                  aria-hidden="true"
                  aria-invalid="false"
                  class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                  name="city"
                  tabindex="-1"
                  value="Kyiv"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-14lo706"
                  >
                    <span>
                      City*
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="checkout-street"
              >
                Street*
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1dnlmzz-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-describedby="helper-text"
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="checkout-street"
                  name="street"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-yjsfm1"
                  >
                    <span>
                      Street*
                    </span>
                  </legend>
                </fieldset>
              </div>
              <p
                class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
                id="helper-text"
              />
            </div>
            <div
              class="MuiBox-root css-l7qti"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="checkout-house"
                >
                  House*
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1dnlmzz-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-describedby="helper-text"
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    id="checkout-house"
                    name="house"
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-yjsfm1"
                    >
                      <span>
                        House*
                      </span>
                    </legend>
                  </fieldset>
                </div>
                <p
                  class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
                  id="helper-text"
                />
              </div>
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="checkout-apartment"
                >
                  Apartment
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1dnlmzz-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-describedby="helper-text"
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    id="checkout-apartment"
                    name="apartment"
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-yjsfm1"
                    >
                      <span>
                        Apartment
                      </span>
                    </legend>
                  </fieldset>
                </div>
                <p
                  class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
                  id="helper-text"
                />
              </div>
            </div>
            <hr
              class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
            />
            <h2
              class="MuiTypography-root MuiTypography-h3 MuiTypography-alignLeft css-1nzy6yt-MuiTypography-root"
            >
              Payment method
            </h2>
            <div
              class="MuiFormControl-root css-1yqd45w-MuiFormControl-root"
            >
              <div
                class="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
                role="radiogroup"
              >
                <label
                  class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-1u6zfwf-MuiFormControlLabel-root"
                >
                  <span
                    class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary Mui-checked MuiRadio-root MuiRadio-colorPrimary css-vqmohf-MuiButtonBase-root-MuiRadio-root"
                  >
                    <input
                      checked=""
                      class="PrivateSwitchBase-input css-1m9pwf3"
                      name="payment"
                      type="radio"
                      value="Card"
                    />
                    <span
                      class="css-hyxlzm"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                        data-testid="RadioButtonUncheckedIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                        />
                      </svg>
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-11zohuh-MuiSvgIcon-root"
                        data-testid="RadioButtonCheckedIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </span>
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                  >
                    Card
                  </span>
                </label>
                <label
                  class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-1u6zfwf-MuiFormControlLabel-root"
                >
                  <span
                    class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-vqmohf-MuiButtonBase-root-MuiRadio-root"
                  >
                    <input
                      class="PrivateSwitchBase-input css-1m9pwf3"
                      name="payment"
                      type="radio"
                      value="Cash"
                    />
                    <span
                      class="css-hyxlzm"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                        data-testid="RadioButtonUncheckedIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                        />
                      </svg>
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
                        data-testid="RadioButtonCheckedIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </span>
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                  >
                    Cash
                  </span>
                </label>
              </div>
            </div>
          </div>
          <div
            class="MuiBox-root css-qga54k"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1x2zd9l-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Back
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-r27lj0-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="submit"
            >
              Continue
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
